File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []
        res = [root.val]
        temp = [root]
        while len(temp):
            arr = []
            for node in temp:
                if node.left:
                    arr.append(node.left)
                if node.right:
                    arr.append(node.right)
            if len(arr):
                res.append(arr[-1].val)
            temp = arr
        return res    

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var rightSideView = function(root) {
    if (!root) return [];
    const res = [root.val];
    let temp = [root];
    while (temp.length) {
        const arr = [];
        for (const node of temp) {
            if (node.left) arr.push(node.left);
            if (node.right) arr.push(node.right);
        }
        if (arr.length) res.push(arr[arr.length - 1].val);
        temp = arr;
    }
    return res;
};